### PR TITLE
Writing Rework

### PR DESCRIPTION
**Pull request by JohnsonArnek**
Reworked writing system added a way for reactionary rebels to push back writing. (only for countries that start wtihout writing)

### CODE REVIEW

**Comment by JohnsonArnek** [comment]
This is completed someone should check over the code though.

File: TGC/common/event_modifiers.txt [hunk 1]
```diff
@@ -5795,4 +5795,87 @@ bad_immigration_policy = {
 	assimilation_rate = -0.05
 	global_immigrant_attract = -0.40
 	icon = 32
+}
+
+Adopted_latin ={
```
Comment (GeneralVonGelre) [inline TGC/common/event_modifiers.txt]: Space missing

File: TGC/common/event_modifiers.txt [hunk 2]
```diff
@@ -5795,4 +5795,87 @@ bad_immigration_policy = {
...
+	global_pop_militancy_modifier = 0.02
+	global_pop_consciousness_modifier = 0.02
+	assimilation_rate = 0.025
+	global_immigrant_attract = 0.02
+	research_points_modifier = 0.06
+	education_efficiency_modifier = 0.1
+	icon = 9
+}
+
+Adopted_Cyrilic =
```
Comment (GeneralVonGelre) [inline TGC/common/event_modifiers.txt]: Does not adhere to code-format of the rest of the mod, = { should be on 1 line

File: TGC/common/event_modifiers.txt [hunk 3]
```diff
@@ -5795,4 +5795,87 @@ bad_immigration_policy = {
...
+	global_pop_consciousness_modifier = 0.01
+	assimilation_rate = 0.025
+	global_immigrant_attract = 0.01
+	research_points_modifier = 0.04
+	war_exhaustion = -0.01
+	education_efficiency_modifier = 0.1
+	icon = 9
+}
+
+Adopted_Hellenic =
```
Comment (GeneralVonGelre) [inline TGC/common/event_modifiers.txt]: See above

File: TGC/common/event_modifiers.txt [hunk 4]
```diff
@@ -5795,4 +5795,87 @@ bad_immigration_policy = {
...
+	global_pop_militancy_modifier = 0.01
+	global_pop_consciousness_modifier = 0.01
+	assimilation_rate = 0.025
+	research_points_modifier = 0.04
+	prestige = 0.01
+	education_efficiency_modifier = 0.1
+	icon = 9
+}
+
+Adopted_Arabic =
```
Comment (GeneralVonGelre) [inline TGC/common/event_modifiers.txt]: See above

File: TGC/common/event_modifiers.txt [hunk 5]
```diff
@@ -5795,4 +5795,87 @@ bad_immigration_policy = {
...
+	global_pop_militancy_modifier = 0.01
+	global_pop_consciousness_modifier = 0.01
+	assimilation_rate = 0.025
+	research_points_modifier = 0.03
+	global_pop_growth = 0.001
+	education_efficiency_modifier = 0.1
+	icon = 9
+}
+
+Adopted_Ethiopic =
```
Comment (GeneralVonGelre) [inline TGC/common/event_modifiers.txt]: See above

File: TGC/common/event_modifiers.txt [hunk 6]
```diff
@@ -5795,4 +5795,87 @@ bad_immigration_policy = {
...
+	global_pop_militancy_modifier = 0.01
+	global_pop_consciousness_modifier = 0.01
+	assimilation_rate = 0.025
+	research_points_modifier = 0.02
+	education_efficiency_modifier = 0.1
+	prestige = 0.02
+	icon = 9
+}
+
+Adopted_Chinese =
```
Comment (GeneralVonGelre) [inline TGC/common/event_modifiers.txt]: See above

File: TGC/common/event_modifiers.txt [hunk 7]
```diff
@@ -5795,4 +5795,87 @@ bad_immigration_policy = {
...
+{
+	global_pop_militancy_modifier = 0.01
+	global_pop_consciousness_modifier = 0.01
+	assimilation_rate = 0.05
+	research_points_modifier = 0.03
+	education_efficiency_modifier = 0.1
+	icon = 9
+}
+
+Adopted_Original =
```
Comment (GeneralVonGelre) [inline TGC/common/event_modifiers.txt]: See above

File: TGC/common/event_modifiers.txt [hunk 8]
```diff
@@ -5795,4 +5795,87 @@ bad_immigration_policy = {
...
+	global_pop_militancy_modifier = 0.01
+	global_pop_consciousness_modifier = 0.01
+	core_pop_militancy_modifier = -0.02
+	assimilation_rate = 0.025
+	research_points_modifier = 0.01
+	education_efficiency_modifier = 0.2
+	prestige = 0.04
+	icon = 9
+}
+
```
Comment (GeneralVonGelre) [inline TGC/common/event_modifiers.txt]: See above

File: TGC/common/event_modifiers.txt [hunk 9]
```diff
@@ -5795,4 +5795,87 @@ bad_immigration_policy = {
...
+Anti_Literacy_Movement =
```
Comment (GeneralVonGelre) [inline TGC/common/event_modifiers.txt]: See above

File: TGC/common/rebel_types.txt [hunk 10]
```diff
@@ -4331,7 +4331,8 @@ unciv_reactionary_rebels = {
 		remove_random_economic_reforms = 1
 		remove_random_military_reforms = 1
 		country_event = 800149
-	}
+		set_country_flag = checker_to_remove_writing
+		}
```
Comment (GeneralVonGelre) [inline TGC/common/rebel_types.txt]: Wrong indentation

File: TGC/decisions/UncivFlavor.txt [hunk 11]
```diff
@@ -5089,5 +5089,31 @@ political_decisions = {
...
+			remove_country_modifier = lacks_writing_system
+			set_country_flag = choose_writing_system
+			set_country_flag = first_writing_group
+			set_country_flag = writing_angery_starter
+			set_country_flag = writing_unciv #Unciv Rebel knower
+			treasury = -10000
+		}
+		ai_will_do = { factor = 1 }
+	}
+	}
```
Comment (GeneralVonGelre) [inline TGC/decisions/UncivFlavor.txt]: Wrong indentation

File: TGC/events/AfricanUncivsFlavour.txt [hunk 12]
```diff
@@ -384,9 +384,242 @@ country_event = {
...
-			# factor = 0
-		# }
-	# }
+		# 
+		#ai_chance = {
+			#factor = 0
+		#}
+	#}
 # }
+
```
Comment (GeneralVonGelre) [inline TGC/events/AfricanUncivsFlavour.txt]: Entire event has wrong indentation

File: TGC/localisation/00_modifiers.csv [hunk 13]
```diff
@@ -1497,3 +1497,19 @@ research_campaign;Active Research Campaign;;;;;;;;;;;;;x
...
 church_property_confiscation_desc;We are fixing our budget holes with the property of the Catholic Church and its monasteries;;;;;;;;;;;;;x
+Adopted_latin;Adopted Latin;;;;;;;;;;;;;x
+Adopted_latin_desc;Adopted the Latin alphabet;;;;;;;;;;;;;x
+Adopted_Cyrilic;Adopted Cyrilic;;;;;;;;;;;;;x
+Adopted_Cyrilic_desc;Adopted the Cyrilic alphabet;;;;;;;;;;;;;x
+Adopted_Hellenic;Adopted Hellenic;;;;;;;;;;;;;x
+Adopted_Hellenic_desc;Adopted the Hellenic alphabet;;;;;;;;;;;;;x
+Adopted_Arabic;Adopted Arabic;;;;;;;;;;;;;x
+Adopted_Arabic_desc;Adopted the Arabic system;;;;;;;;;;;;;x
+Adopted_Ethiopic;Adopted Ethiopic;;;;;;;;;;;;;x
```
Comment (GeneralVonGelre) [inline TGC/localisation/00_modifiers.csv]: Ethiopic is not a word, should be Ethiopian

File: TGC/localisation/00_modifiers.csv [hunk 14]
```diff
@@ -1497,3 +1497,19 @@ research_campaign;Active Research Campaign;;;;;;;;;;;;;x
...
+Adopted_Ethiopic_desc;Adopterd the Ethiopic system;;;;;;;;;;;;;x
+Adopted_Chinese;Adopted Chinese;;;;;;;;;;;;;x
+Adopted_Chinese_desc;Adopted the Chinese system;;;;;;;;;;;;;x
+Adopted_Original;Adopted Original;;;;;;;;;;;;;x
+Adopted_Original_desc;Adopted the Original writing system;;;;;;;;;;;;;x
```
Comment (GeneralVonGelre) [inline TGC/localisation/00_modifiers.csv]: What is "The original" writing system? Cave-drawings?

**Review by GeneralVonGelre — CHANGES_REQUESTED** [review]

File: TGC/localisation/00_modifiers.csv (hunk 13 above)
Comment (JohnsonArnek) [reply TGC/localisation/00_modifiers.csv]: Ethiopian Semitic (also Ethio-Semitic, Ethiosemitic, Ethiopic or Abyssinian[[1]](https://en.wikipedia.org/wiki/Ethiopian_Semitic_languages#cite_note-Z2-5-1)) i

File: TGC/localisation/00_modifiers.csv (hunk 14 above)
Comment (JohnsonArnek) [reply TGC/localisation/00_modifiers.csv]: Original can be anything from an Original: logographic, syllabic, alphabetic, and featural system.

Comment (JohnsonArnek) [reply TGC/localisation/00_modifiers.csv]: Arguably the Korean Alphabet is an Original writing system that has come about thanks to western expansion

**Comment by JohnsonArnek** [comment]
The Greek Language Debate Added.

**Comment by JohnsonArnek** [comment]
Added an Easter Egg Event

**Comment by JohnsonArnek** [comment]
Added Greek Alternative Paths

**Pull request closed without merging**